// @wailkouicem/common/build/events/listeners/base-listner-class.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Listener = void 0;
class Listener {
    //constructor -----------------------------------------
    constructor(client) {
        this.ackWait = 5 * 1000;
        this.client = client;
    }
    //helpers ---------------------------------------
    subscriptionOptions() {
        return this.client.subscriptionOptions()
            .setManualAckMode(true)
            .setDeliverAllAvailable()
            .setAckWait(this.ackWait)
            .setDurableName(this.queueGroupName);
    }
    parseMessage(msg) {
        const data = msg.getData();
        if (typeof data === "string") {
            return JSON.parse(data);
        }
        else {
            return "";
        }
    }
    //main functions ------------------------------------
    listen() {
        const subscribtion = this.client.subscribe(this.subject, this.queueGroupName, this.subscriptionOptions());
        console.log("lisetening to ", this.subject, " chanel");
        subscribtion.on("message", (msg) => {
            console.log(msg.getSequence(), " ", this.subject, " ", this.queueGroupName);
            const parsedData = this.parseMessage(msg);
            this.onMessage(msg, parsedData);
        });
    }
}
exports.Listener = Listener;
